pkg/ccl/utilccl/license_check.go

```diff
@@ -76,6 +78,28 @@ func CheckEnterpriseEnabled(st *cluster.Settings, cluster uuid.UUID, org, featur
 func init() {
 	base.CheckEnterpriseEnabled = CheckEnterpriseEnabled
 	base.LicenseType = getLicenseType
+	base.TimeToLicenseExpiry = TimeToLicenseExpiry
+}
+
+func TimeToLicenseExpiry(ctx context.Context, st *cluster.Settings) (time.Duration, error) {
+	var lic *licenseccl.License
+	// FIXME(tschottdorf): see whether it makes sense to cache the decoded
+	// license.
+	// TODO(davidh): perhaps above comment is extremely important if this is recomputed often.
+	if str := enterpriseLicense.Get(&st.SV); str != "" {
+		var err error
+		if lic, err = decode(str); err != nil {
+			return 0, err
+		}
+	}
+
+	if lic == nil {
+		return 0, errors.New("unable to compute expiry")
```

Comment:
I'm not sure this is an error? we already return a `decode` error so this is only nil if `str` above is "", and that means there is no license. I'd might just return `0, nil` if str is "", then eliminate this check?